pkg/lib/bundle/validate.go

```diff
@@ -0,0 +1,147 @@
+type ImageValidator struct {
+	ImageReader containertools.ImageReader
+	Logger      *log.Entry
+}
+
+// PullBundleImage shells out to a container tool and pulls a given image tag
+// Then it unpacks the image layer filesystem contents and pushes the contents
+// to a specified directory for further validation
+func (i ImageValidator) PullBundleImage(tag, dir string) error {
+	i.Logger.Debug("Pulling and unpacking container image")
+
+	err := i.ImageReader.GetImageData(tag, dir)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateBundle takes a directory containing the contents of a bundle and validates
+// the format of that bundle for correctness
+func (i ImageValidator) ValidateBundle(dir string) error {
+	var manifestsFound, metadataFound bool
+	var annotationsDir, manifestsDir string
+	var annotationErrors []error
+	var formatErrors []error
+
+	items, _ := ioutil.ReadDir(dir)
+	for _, item := range items {
+		if item.IsDir() {
+			switch s := item.Name(); s {
+			case strings.TrimSuffix(ManifestsDir, "/"):
+				i.Logger.Debug("Found manifests directory")
+				manifestsFound = true
+				manifestsDir = filepath.Join(dir, ManifestsDir)
+			case strings.TrimSuffix(MetadataDir, "/"):
+				i.Logger.Debug("Found metadata directory")
+				metadataFound = true
+				annotationsDir = filepath.Join(dir, MetadataDir)
+			}
+		}
+	}
+
+	if manifestsFound == false {
+		formatErrors = append(formatErrors, fmt.Errorf("Unable to locate manifests directory"))
+	}
+	if metadataFound == false {
+		formatErrors = append(formatErrors, fmt.Errorf("Unable to locate metadata directory"))
+	}
+
+	// Break here if we can't even find the files
+	if len(formatErrors) > 0 {
+		return NewValidationError(annotationErrors, formatErrors)
+	}
+
+	i.Logger.Debug("Getting mediaType info from manifests directory")
+	mediaType, err := GetMediaType(manifestsDir)
+	if err != nil {
+		formatErrors = append(formatErrors, err)
+	}
+
+	// Validate annotations.yaml
+	annotationsFile, err := ioutil.ReadFile(filepath.Join(annotationsDir, AnnotationsFile))
+	if err != nil {
+		fmtErr := fmt.Errorf("Unable to read annotations.yaml file: %s", err.Error())
+		formatErrors = append(formatErrors, fmtErr)
+	}
+
+	var fileAnnotations AnnotationMetadata
+
+	annotations := map[string]string{
+		MediatypeLabel:      mediaType,
+		ManifestsLabel:      ManifestsDir,
+		MetadataLabel:       MetadataDir,
+		PackageLabel:        "",
+		ChannelsLabel:       "",
+		ChannelDefaultLabel: "",
+	}
+
+	i.Logger.Debug("Validating annotations.yaml")
+
+	err = yaml.Unmarshal(annotationsFile, &fileAnnotations)
+	if err != nil {
+		formatErrors = append(formatErrors, fmt.Errorf("Unable to parse annotations.yaml file"))
+	}
+
+	for label, item := range annotations {
+		val, ok := fileAnnotations.Annotations[label]
+		if ok {
+			i.Logger.Debugf(`Found annotation "%s" with value "%s"`, label, val)
+		} else {
+			aErr := fmt.Errorf(`Missing annotation "%s"`, label)
```

Comment:
Nit: You can use `%q` to quote strings in format strings:
```suggestion
			aErr := fmt.Errorf("Missing annotation %q", label)
```